lib/bolt/module_installer/puppetfile.rb

```diff
@@ -97,20 +97,43 @@ def assert_satisfies(specs)
           end
         end
 
-        return if unsatisfied_specs.empty?
-
+        #versionless_mods = @modules.any? do |mod|    
+        #  mod.version.nil? 
+        #end
+        versionless_mods = []
+        @modules.each {|mod|
+          if mod.version.nil?
+            versionless_mods << mod
+          end
+        }
```

Comment:
On lines 100-102, I attempted to do the pretty ruby version of what I do on 103-107. Syntax help would be appreciated :)